xknx/management/procedures.py

```diff
@@ -184,3 +184,58 @@ async def nm_invididual_address_write(
             tpci=tpci.TDataConnected(sequence_number=seq_num),
         )
         await xknx.cemi_handler.send_telegram(telegram)
+
+
+async def nm_individual_address_serial_number_read(
+    xknx: XKNX, serial: bytes, timeout: float = 3
+) -> IndividualAddress | None:
+    """Read individual address from device with specified serial number."""
+
+    # initialize queue or event handler gathering broadcasts
+    async with xknx.management.broadcast() as bc_context:
+        broadcast_telegram = Telegram(
+            destination_address=GroupAddress("0/0/0"),
+            source_address=xknx.current_address,
+            payload=apci.IndividualAddressSerialRead(serial=serial),
+        )
+        await xknx.management.send_broadcast(broadcast_telegram)
+        async for result in bc_context.receive(timeout=timeout):
+            if isinstance(result.payload, apci.IndividualAddressSerialResponse):
+                if result.serial != serial:
+                    raise ManagementConnectionError("Transmitted and received serial numbers do not match.")
```

Comment:
Do we need to `raise` or simply ignore this response?

Reply:
I would raise an error, since there was a response, but not the expected one. The correct response might still appear on the bus later. If we return None, the issuer might incorrectly conclude that the device is not present on the bus. If we raise an error, the issuer can reattempt to access the individual address of the device.

Reply:
I would argue the opposite and simply ignore if we are not the destination (which we know), the caller should not need to care and wait for a timeout IMO.

Reply:
Fair point, I updated the code to ignore other broadcast messages. A timeout error will not be thrown as passed to the management procedure. I also modified the tests and error messages reported to better reflect what happened.